text/0000-pack-package-refactor.md

```diff
@@ -0,0 +1,118 @@
+# Meta
+
+- Name: Refactor Pack go packages
+- Start Date: 02/24/21
+- Author(s): dwillist
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: (put "N/A" unless this replaces an existing RFC, then link to that RFC)
+
+# Summary
+[summary]: #summary
+
+This RFC outlines a promotion strategy and process for refactoring `pack` into consumable components as needed.
+
+# Motivation
+[motivation]: #motivation
+
+- Why should we do this?
+    - Establish some shared vision about the codebase we want to manage.
+    - General refactoring benifits: 
+        - Maintaining a modular and re-usable codebase enables contributors to be more effective, and encourages new adopters.
+        - Ease consumption of pack as a library by having isolated and focused packages
+        - Create logical seams to redistribute packages.
+
+# What it is
+[what-it-is]: #what-it-is
+The general flow of this process is
+1) Unorganized features exist & are added to the `internal` package, these are not consumable by external programs
+2) Extract code from the `interal` package to `/pkg/<pkg-name>/...` to enable external consumption.
+3) Provide some useful set of functionality through independent binaries at `/cmd/<bin-name>` that other platforms may consume.
+4) Extract code from external packages `/pkg/<pkg-name>/...` into a separate repository (this should require an RFC)
+
+The goal of this RFC is to:
+- Reach consensus on the overall code promotion process for the `pack` repo
+- Reach consensus on a "probable" target structure for `pack`
+
+The goal of this RFC is NOT:
+- to precisely define what packages we want & what their contents/functionality should be.
+
+# How it Works
+[how-it-works]: #how-it-works
+
+### Current structure
+The current state of the `pack` repo does not provide a standardized structure, and much of the logic
+in the inaccessible `internal` package.
+```
+.
+├── builder
+├── buildpackage
+├── cmd
+│    └── pack
+├── config
+├── internal/...
+├── logging
+├── out
+│    └── tests
```

Comment:
```suggestion
```

These don't exist by default, they are just created through the testing structure.